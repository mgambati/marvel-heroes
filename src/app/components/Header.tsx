import React from "react";
import { Container, Flex, Text, Image, Link as LinkThemeUI } from "theme-ui";
import logo from "../assets/logo.png";
import { Link } from "react-router-dom";

export const Header = () => {
  return (
    <Flex
      as="header"
      sx={{
        justifyContent: "center",
        borderBottom: "1px solid",
        borderColor: "gray.4",
      }}
    >
      <Container sx={{ py: 3, maxWidth: "64em" }}>
        <Flex sx={{ justifyContent: "space-between", alignItems: "center" }}>
          <Flex>
            <Link to="/">
              <Image sx={{ maxWidth: "100px" }} src={logo} />
            </Link>
          </Flex>

          <Flex
            sx={{
              flexDirection: "column",
              textAlign: "right",
            }}
          >
            <Text>Matheus Gambati</Text>
            <LinkThemeUI href="https://github.com/mgambati">Github</LinkThemeUI>
            <LinkThemeUI href="https://www.linkedin.com/in/matheusgambati/">
              LinkedIn
            </LinkThemeUI>
          </Flex>
        </Flex>
      </Container>
    </Flex>
  );
};
