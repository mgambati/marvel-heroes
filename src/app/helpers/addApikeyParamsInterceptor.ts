import { AxiosRequestConfig } from "axios";

/**
 * Devido a um bug no Axios, default params n√£o sofre merge com params da request;
 * Para resolver isso usa-se um interceptor
 */
export function addApikeyParamsInterceptor(
  config: AxiosRequestConfig
): AxiosRequestConfig {
  config.params = {
    ...config.params,
    apikey: process.env["REACT_APP_MARVEL_API_KEY"],
  };

  return config;
}
