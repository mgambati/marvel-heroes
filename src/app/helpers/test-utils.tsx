import { render as rtlRender } from "@testing-library/react";
import React, { FC } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider, Theme } from "theme-ui";
import { ThunkDispatch, AnyAction } from "@reduxjs/toolkit";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import { RootState, store } from "../store";
import { BASE_THEME } from "../constants/BASE_THEME";

export function setupMockedStore() {
  type DispatchExts = ThunkDispatch<RootState, void, AnyAction>;
  const middlewares = [thunk];
  const mockStore = configureStore<RootState, DispatchExts>(middlewares);
  const initialState: RootState = {};

  return [mockStore, initialState] as [typeof mockStore, typeof initialState];
}

function render(
  ui: JSX.Element,
  { initialState = undefined, mockedStore = store, ...renderOptions } = {}
) {
  const Wrapper: FC = ({ children }) => {
    return (
      <ThemeProvider theme={BASE_THEME as Theme}>
        <BrowserRouter>
          <Provider store={store}>{children}</Provider>
        </BrowserRouter>
      </ThemeProvider>
    );
  };

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

export { render };
