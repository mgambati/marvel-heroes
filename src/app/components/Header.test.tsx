import React from "react";
import { Header } from "./Header";
import { render } from "../helpers/test-utils";

test("renders correctly", () => {
  const { container } = render(<Header />);

  expect(container).toMatchSnapshot();
});
