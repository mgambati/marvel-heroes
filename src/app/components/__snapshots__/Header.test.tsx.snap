// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    class="css-1meb1r7"
  >
    <div
      class="css-1n1y94f"
    >
      <div
        class="css-199kefw"
      >
        <div
          class="css-4cffwv"
        >
          <a
            href="/"
          >
            <img
              class="css-1u1uv2u"
              src="logo.png"
            />
          </a>
        </div>
        <div
          class="css-1cwf0tq"
        >
          <div
            class="css-10hm5fg"
          >
            Matheus Gambati
          </div>
          <a
            class="css-1eyjkyt"
            href="https://github.com/mgambati"
          >
            Github
          </a>
          <a
            class="css-1eyjkyt"
            href="https://www.linkedin.com/in/matheusgambati/"
          >
            LinkedIn
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
